### PR TITLE
feat: link to build artefacts on successful CI runs for PR's.

### PR DESCRIPTION
**Pull request by sysvinit**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When new UI elements are added, they need to pass through a design review process, for which the Android team needs to provide an updated APK to the Design team. CI is configured to run automatically when pull requests are opened or updated on GitHub, which includes uploading the APK produced by the CI build to an S3 bucket. @MohamadJaara has requested that when CI builds a client from a PR branch successfully, a link to the build in S3 should automatically be posted in a comment on GitHub, to make it clearer which build is generated by a given PR.

### Solutions

I have modified the Jenkins CI pipeline to post a comment to GitHub with a link to the client build on S3 when builds on PR branches complete successfully.

### Testing

Assuming I understand how CI works correctly, submitting this PR should trigger a build based on the modified `Jenkinsfile` in this branch, which should cause a comment to be posted on this PR if the build succeeds.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by sysvinit** [comment]
https://xkcd.com/1638/

xkcd 1638

File: Jenkinsfile [hunk 1]
```diff
@@ -342,6 +368,13 @@ pipeline {
 
   post {
     failure {
+      script {
+        if (env.BRANCH_NAME.startsWith('PR-')) {
```
Comment (MohamadJaara) [inline Jenkinsfile:372]: Is there a reason for posting GitHub comment only if the branch name start with  PR-

**Comment by AndroidBob** [comment]
Build [16](https://10.10.124.134/job/wire-android-reloaded/job/PR-458/16/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(2975609)(2975609)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(2975609)(2975609)-dev-release.apk)

**Review by vitorhugods — APPROVED** [review]
👨🏼‍🍳🤌🏼

**Review by MohamadJaara — APPROVED** [review]
Nicely done 🚀

**Pull request merged**